### DISCUSSION TITLE
Using `authField` as well as other methods

### DISCUSSION THREAD

**Discussion by axhamre** [hayes/pothos#422] [Q&A] [answered]
I'm using `authField` (succesfully) to change context types [based on scopes](https://pothos-graphql.dev/docs/plugins/scope-auth#change-context-types-based-on-scopes). However it's position is in place of other methods like `prismaField` and `prismaConnection` that I'd like to use as well.

Is there any example how `authField` can be used in conjunction with any of the example methods above?

**Comment by hayes** [comment, marked as answer]
Edit: A new method `t.withAuth` has been added to help with this issue.

Unfortunately there isn't a good way to do this right now...  I have tried to make plugins compatible as much as possible, but this is a feature that slipped through the cracks, and doesn't have a good way to work together.  For now the best work around is to use t.prismaField with the `auth` option for what ever scopes you want to apply, and then use a type-cast on the context object when accessing it inside your resolver.

I am open to ideas on how this could work better.  Unfortunately it isn't possible to just change the type for context automatically in any field builder method when the auth option is defined, and creating an auth plugin version of each method is probably too much overhead   I'm not sure if it's possible, but it might be possible to create something like `t.withAuth({ scopes: ... }).prismaField(...)`, but getting that to work correctly would probably be fairly involved.

**Reply by axhamre** [reply]
Oh I see. I don't have any suggestion myself, but I think `withAuth` would be sweet if it worked - it's probably a common case not wanting to non-null assert the context properties for every private/protected query and mutation.

**Reply by hayes** [reply]
Implementing this here: https://github.com/hayes/pothos/pull/438

**Reply by axhamre** [reply]
Wow, amazing! And the method chaining looks great IMO.

**Reply by hayes** [reply]
This should now be available in the latest version.
